### PR TITLE
#3757 update Fela example for Next 5

### LINKED ISSUE #3757: Fela.js integration is broken after 5.0 (with-fela example)
- [x] I have searched the [issues](https://github.com/zeit/next.js/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior
Expected to render css on the server first, using specific code in `_document.js`. So, expected behavior would be to see no FOUC.

## Current Behavior
CSS is not rendered on the server, only on the client. It's interesting that class names are applied correctly on the server, only actual CSS is not rendered. As the result, we see FOUC before client side catch up.

This happened right after updated to 5.0. Latest `canary` releases also broken.

## Steps to Reproduce (for bugs)
1. `curl https://codeload.github.com/zeit/next.js/tar.gz/canary | tar -xz --strip=2 next.js-canary/examples/with-fela`
2. `cd with-fela`
3. `yarn`
4. `yarn dev`

## Your Environment
| Tech    | Version |
|---------|---------|
| next    | 5.0.0 / 5.0.1-canary.4 / 5.0.1-canary.5 / 5.0.1-canary.9        |
| node    | 9.5.0        |
| OS      |  OSX 10.13.3       |

### PR DESCRIPTION
**Pull request by robinweser**
closes #3757

### CODE REVIEW

File: examples/with-fela/pages/_document.js [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 export default class MyDocument extends Document {
   static getInitialProps ({ renderPage }) {
-    const page = renderPage()
-    const sheetList = renderToSheetList(felaRenderer)
-    felaRenderer.clear()
+    const serverRenderer = getFelaRenderer()
+
+    const page = renderPage(App => props => (
+      <FelaProvider renderer={serverRenderer}>
+        <App />
```
Comment (timneutkens) [inline examples/with-fela/pages/_document.js]: I'm pretty sure we don't pass props, but to make sure it keeps working in the future:

```
<App {...props} />
```

File: lib/head.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ function unique () {
   const metaCategories = {}
 
   return (h) => {
-    if (h.key && h.key.startsWith('.$')) {
+    if (h.key && h.key.indexOf('.$') === 0) {
```
Comment (timneutkens) [inline lib/head.js]: I wonder where this came from. But it should definitely be added. 👍

**Comment by timneutkens** [comment]
Nice @rofrischmann 👍 Just had one comment.

**Comment by rofrischmann** [comment]
Heyhey, sorry for the mess. Opened a new PR using the canary as base branch so only my changes are merged :P 
See https://github.com/zeit/next.js/pull/3949

**Pull request closed without merging**